### PR TITLE
Authentication

### LINKED ISSUE #124: Web Form for Sign in (authorization)
Form with an implemented button for further Sign up (registration)

### PR DESCRIPTION
**Pull request by radik1999**

### CODE REVIEW

**Review by AndriiSkavreniuk — APPROVED** [review]

File: library/catalog/models.py [hunk 1]
```diff
@@ -1,7 +1,12 @@
 from datetime import datetime
+
+from django.contrib.auth import get_user_model
 from django_mongoengine import Document, EmbeddedDocument
 from mongoengine import DateTimeField, EmailField, EmbeddedDocumentField, FloatField, \
     IntField, ListField, ReferenceField, StringField
+from werkzeug.security import check_password_hash, generate_password_hash
```
Comment (ivan-dubei) [inline library/catalog/models.py:9]: alphabetical order

File: library/catalog/views.py [hunk 2]
```diff
@@ -1,18 +1,12 @@
+from django.contrib.auth.models import User
 from django.http import HttpResponse
 from django.shortcuts import redirect, render
 from mongoengine.queryset.visitor import Q
-from werkzeug.security import generate_password_hash
+from django.contrib.auth import login, authenticate, logout
```
Comment (ivan-dubei) [inline library/catalog/views.py]: Alphabetical order

File: library/library/settings.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
 
 from dotenv import load_dotenv
 from pathlib import Path
+import mongoengine
```
Comment (ivan-dubei) [inline library/library/settings.py]: wrong order
usually we put 'import something' at the first place

File: library/library/settings.py [hunk 4]
```diff
@@ -27,7 +28,7 @@
 SECRET_KEY = os.getenv('DJANGO_SECRET_KEY')
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = False
+DEBUG = True
```
Comment (ivan-dubei) [inline library/library/settings.py]: Is it possible to move this value into config file ?

File: library/library/urls.py [hunk 5]
```diff
@@ -14,7 +14,9 @@
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
 from django.urls import include, path
+from django.contrib import admin
```
Comment (ivan-dubei) [inline library/library/urls.py:16]: alphabetical order

File: library/catalog/models.py [hunk 6]
```diff
@@ -67,7 +104,7 @@ class Book(Document):
     statistic = EmbeddedDocumentField(BookStatistic.__name__, default=BookStatistic())
```
Comment (AndriiSkavreniuk) [inline library/catalog/models.py:100]: add a blank line

**Review by ivan-dubei — APPROVED** [review]

**Review by mehalyna — APPROVED** [review]

**Pull request merged**